lib/Rest/HttpRequest.cpp

```diff
@@ -42,6 +43,30 @@
 using namespace arangodb;
 using namespace arangodb::basics;
 
+namespace {
+std::string url_decode(const char* begin, const char* end) {
+  std::string out;
+  out.reserve(static_cast<size_t>(end - begin));
+  for (const char* i = begin; i != end; ++i) {
```

Comment:
This looks familiar.
We have the same code in fuerte messages.
As fuerte is "3rdParty" i am open to not unify this, though I would prefer a single implementation again.
However the same question holds here, how save are we against crafted invalid URLs?

Reply:
Ok, I will make it throw an exception in case of invalid encodings used in URLs